syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -21,8 +24,35 @@ def __private_unfold(image, kernel_size, padding):
         return image.unfold(kernel_size, padding)
 
     @staticmethod
-    def xor(value, other):
-        return value + other - (value * 2 * other)
+    def xor(
+        value: ReplicatedSharingTensor, other: Union[int, ReplicatedSharingTensor, torch.LongTensor]
+    ) -> ReplicatedSharingTensor:
+        """
+        Compute the XOR value between value and other.
+        If value and other are both ints we should use the "^" operator.
+
+        Args:
+            value (ReplicatedSharingTensor): RST with ring size of 2 or integer value in {0, 1}
+            other (int): integer with value in {0, 1}
+
+        Returns:
+            The XOR computation between value and other
+        """
+
+        if not isinstance(value, ReplicatedSharingTensor) or value.ring_size != 2:
```

Comment:
I have all those types in place - which look pretty bad - let's keep them in place only to make sure we define the protocol correctly and then remove them